crud/select/plan.lua

```diff
@@ -106,10 +107,70 @@ local function extract_sharding_key_from_scan_value(scan_value, scan_index, shar
     return sharding_key
 end
 
+-- We need to construct after_tuple by field_names
+-- because if `fields` option is specified we have after_tuple with partial fields
+-- and these fields are ordered by field_names + primary key + scan key
+-- this order can be differ from order in space format
+-- so we need to cast after_tuple to space format for scrolling tuples on storage
+local function construct_after_tuple_by_fields(space_format, field_names, tuple)
+    if tuple == nil then
+        return nil
+    end
+
+    if field_names == nil then
+        return tuple
+    end
+
+    local positions = {}
+    local transformed_tuple = {}
+
+    for i, field in ipairs(space_format) do
+        positions[field.name] = i
+    end
+
+    for i, field_name in ipairs(field_names) do
+        local fieldno = positions[field_name]
+        if fieldno == nil then
+            return nil, FilterFieldsError:new(
+                    'Space format doesn\'t contain field named %q', field_name
+            )
+        end
+
+        transformed_tuple[fieldno] = tuple[i]
+    end
+
+    return transformed_tuple
+end
```

Comment:
This function also needs unit test. It can be tested [here](https://github.com/tarantool/crud/blob/master/test/unit/select_plan_test.lua) as a part of `select.new` tests.

Moreover, now plan contains `field_names`, please add this check to plan tests.